### PR TITLE
test: move help strings to markdown file

### PR DESCRIPTION
**Pull request by m11o**
issue: #4368

This PR contains bellow contents.

- Remove duplicated expressions 22d1880e715ce6e224c19c2cc3e3c2aeaea92822
- Move help strings to markdown file d6366c41eda316cad6b1c1047d97bed9e2b4b078

This command has two util names ( `test` or `[` ). And, the format_usage method replaces {} with a util name. So, we changed `USAGE` as strings that do not contain `{}`.
If there are better ideas, please tell me.

## before

```
$ cargo run -- [ --help
Check file types and compare values.

Usage: target/debug/coreutils [ EXPRESSION
           target/debug/coreutils [
           [ EXPRESSION ]
           [ ]
           [ OPTION

Options:
  -h, --help     Print help
  -V, --version  Print version


Exit with the status determined by EXPRESSION.

An omitted EXPRESSION defaults to false.  Otherwise,
EXPRESSION is true or false and sets exit status.  It is one of:

  ( EXPRESSION )               EXPRESSION is true
  ! EXPRESSION                 EXPRESSION is false
  EXPRESSION1 -a EXPRESSION2   both EXPRESSION1 and EXPRESSION2 are true
  EXPRESSION1 -o EXPRESSION2   either EXPRESSION1 or EXPRESSION2 is true

  -n STRING            the length of STRING is nonzero
  STRING               equivalent to -n STRING
  -z STRING            the length of STRING is zero
  STRING1 = STRING2    the strings are equal
  STRING1 != STRING2   the strings are not equal

  INTEGER1 -eq INTEGER2   INTEGER1 is equal to INTEGER2
  INTEGER1 -ge INTEGER2   INTEGER1 is greater than or equal to INTEGER2
  INTEGER1 -gt INTEGER2   INTEGER1 is greater than INTEGER2
  INTEGER1 -le INTEGER2   INTEGER1 is less than or equal to INTEGER2
  INTEGER1 -lt INTEGER2   INTEGER1 is less than INTEGER2
  INTEGER1 -ne INTEGER2   INTEGER1 is not equal to INTEGER2

  FILE1 -ef FILE2   FILE1 and FILE2 have the same device and inode numbers
  FILE1 -nt FILE2   FILE1 is newer (modification date) than FILE2
  FILE1 -ot FILE2   FILE1 is older than FILE2

  -b FILE     FILE exists and is block special
  -c FILE     FILE exists and is character special
  -d FILE     FILE exists and is a directory
  -e FILE     FILE exists
  -f FILE     FILE exists and is a regular file
  -g FILE     FILE exists and is set-group-ID
  -G FILE     FILE exists and is owned by the effective group ID
  -h FILE     FILE exists and is a symbolic link (same as -L)
  -k FILE     FILE exists and has its sticky bit set
  -L FILE     FILE exists and is a symbolic link (same as -h)
  -N FILE     FILE exists and has been modified since it was last read
  -O FILE     FILE exists and is owned by the effective user ID
  -p FILE     FILE exists and is a named pipe
  -r FILE     FILE exists and read permission is granted
  -s FILE     FILE exists and has a size greater than zero
  -S FILE     FILE exists and is a socket
  -t FD       file descriptor FD is opened on a terminal
  -u FILE     FILE exists and its set-user-ID bit is set
  -w FILE     FILE exists and write permission is granted
  -x FILE     FILE exists and execute (or search) permission is granted

Except for -h and -L, all FILE-related tests dereference symbolic links.
Beware that parentheses need to be escaped (e.g., by backslashes) for shells.
INTEGER may also be -l STRING, which evaluates to the length of STRING.

NOTE: Binary -a and -o are inherently ambiguous.  Use 'test EXPR1 && test
EXPR2' or 'test EXPR1 || test EXPR2' instead.

NOTE: [ honors the --help and --version options, but test does not.
test treats each of those as it treats any other nonempty STRING.

NOTE: your shell may have its own version of test and/or [, which usually supersedes
the version described here.  Please refer to your shell's documentation
for details about the options it supports.
```

## after

```
$ cargo run -- [ --help
Check file types and compare values.

Usage: test EXPRESSION
       test
       [ EXPRESSION ]
       [ ]
       [ OPTION

Options:
  -h, --help     Print help
  -V, --version  Print version

Exit with the status determined by EXPRESSION.

An omitted EXPRESSION defaults to false.
Otherwise, EXPRESSION is true or false and sets exit status.

It is one of:

( EXPRESSION )               EXPRESSION is true
! EXPRESSION                 EXPRESSION is false
EXPRESSION1 -a EXPRESSION2   both EXPRESSION1 and EXPRESSION2 are true
EXPRESSION1 -o EXPRESSION2   either EXPRESSION1 or EXPRESSION2 is true

-n STRING            the length of STRING is nonzero
STRING               equivalent to -n STRING
-z STRING            the length of STRING is zero
STRING1 = STRING2    the strings are equal
STRING1 != STRING2   the strings are not equal

INTEGER1 -eq INTEGER2   INTEGER1 is equal to INTEGER2
INTEGER1 -ge INTEGER2   INTEGER1 is greater than or equal to INTEGER2
INTEGER1 -gt INTEGER2   INTEGER1 is greater than INTEGER2
INTEGER1 -le INTEGER2   INTEGER1 is less than or equal to INTEGER2
INTEGER1 -lt INTEGER2   INTEGER1 is less than INTEGER2
INTEGER1 -ne INTEGER2   INTEGER1 is not equal to INTEGER2

FILE1 -ef FILE2   FILE1 and FILE2 have the same device and inode numbers
FILE1 -nt FILE2   FILE1 is newer (modification date) than FILE2
FILE1 -ot FILE2   FILE1 is older than FILE2

-b FILE     FILE exists and is block special
-c FILE     FILE exists and is character special
-d FILE     FILE exists and is a directory
-e FILE     FILE exists
-f FILE     FILE exists and is a regular file
-g FILE     FILE exists and is set-group-ID
-G FILE     FILE exists and is owned by the effective group ID
-h FILE     FILE exists and is a symbolic link (same as -L)
-k FILE     FILE exists and has its sticky bit set
-L FILE     FILE exists and is a symbolic link (same as -h)
-N FILE     FILE exists and has been modified since it was last read
-O FILE     FILE exists and is owned by the effective user ID
-p FILE     FILE exists and is a named pipe
-r FILE     FILE exists and read permission is granted
-s FILE     FILE exists and has a size greater than zero
-S FILE     FILE exists and is a socket
-t FD       file descriptor FD is opened on a terminal
-u FILE     FILE exists and its set-user-ID bit is set
-w FILE     FILE exists and write permission is granted
-x FILE     FILE exists and execute (or search) permission is granted

Except for -h and -L, all FILE-related tests dereference symbolic links.
Beware that parentheses need to be escaped (e.g., by backslashes) for shells.
INTEGER may also be -l STRING, which evaluates to the length of STRING.

NOTE: Binary -a and -o are inherently ambiguous.
Use 'test EXPR1 && test EXPR2' or 'test EXPR1 || test EXPR2' instead.

NOTE: [ honors the --help and --version options, but test does not.
test treats each of those as it treats any other nonempty STRING.

NOTE: your shell may have its own version of test and/or [, which usually supersedes the version described here.
Please refer to your shell's documentation for details about the options it supports.
```

### CODE REVIEW

**Comment by cakebaker** [comment]
Thanks for your PR. Can you please use some markdown lists? The rendered markdown currently looks a bit messy: https://github.com/uutils/coreutils/blob/7b23281b06011ff7838d8c2cab6bb084fdfc46f4/src/uu/test/test.md

File: src/uu/test/test.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+An omitted `EXPRESSION` defaults to false.
+Otherwise, `EXPRESSION` is true or false and sets exit status. 
+
+It is one of:
+
+( EXPRESSION )               `EXPRESSION` is true
+! EXPRESSION                 `EXPRESSION` is false
+EXPRESSION1 -a EXPRESSION2   both `EXPRESSION1` and `EXPRESSION2` are true
+EXPRESSION1 -o EXPRESSION2   either `EXPRESSION1` or `EXPRESSION2` is true
```
Comment (sylvestre) [inline src/uu/test/test.md:25]:
```suggestion
* ( EXPRESSION )               `EXPRESSION` is true
* ! EXPRESSION                 `EXPRESSION` is false
* EXPRESSION1 -a EXPRESSION2   both `EXPRESSION1` and `EXPRESSION2` are true
* EXPRESSION1 -o EXPRESSION2   either `EXPRESSION1` or `EXPRESSION2` is true
```

File: src/uu/test/test.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+-n STRING            the length of `STRING` is nonzero
+STRING               equivalent to -n `STRING`
+-z STRING            the length of `STRING` is zero
+STRING1 = STRING2    the strings are equal
+STRING1 != STRING2   the strings are not equal
```
Comment (sylvestre) [inline src/uu/test/test.md:31]:
```suggestion
* -n STRING            the length of `STRING` is nonzero
* STRING               equivalent to -n `STRING`
* -z STRING            the length of `STRING` is zero
* STRING1 = STRING2    the strings are equal
* STRING1 != STRING2   the strings are not equal
```

File: src/uu/test/test.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+INTEGER1 -eq INTEGER2   `INTEGER1` is equal to `INTEGER2`
+INTEGER1 -ge INTEGER2   `INTEGER1` is greater than or equal to `INTEGER2`
+INTEGER1 -gt INTEGER2   `INTEGER1` is greater than `INTEGER2`
+INTEGER1 -le INTEGER2   `INTEGER1` is less than or equal to `INTEGER2`
+INTEGER1 -lt INTEGER2   `INTEGER1` is less than `INTEGER2`
+INTEGER1 -ne INTEGER2   `INTEGER1` is not equal to `INTEGER2`
+
+FILE1 -ef FILE2   `FILE1` and `FILE2` have the same device and inode numbers
```
Comment (sylvestre) [inline src/uu/test/test.md:40]: same here and below

File: src/uu/test/test.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+-p FILE     `FILE` exists and is a named pipe
+-r FILE     `FILE` exists and read permission is granted
+-s FILE     `FILE` exists and has a size greater than zero
+-S FILE     `FILE` exists and is a socket
+-t FD       `file` descriptor `FD` is opened on a terminal
+-u FILE     `FILE` exists and its set-user-ID bit is set
+-w FILE     `FILE` exists and write permission is granted
+-x FILE     `FILE` exists and execute (or search) permission is granted
+
+Except for -h and -L, all FILE-related tests dereference symbolic links.
```
Comment (sylvestre) [inline src/uu/test/test.md:65]:
```suggestion
Except for `-h` and `-L`, all FILE-related tests dereference (follow) symbolic links.
```

File: src/uu/test/test.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+Beware that parentheses need to be escaped (e.g., by backslashes) for shells.
+`INTEGER` may also be -l `STRING`, which evaluates to the length of `STRING`.
+
+NOTE: Binary -a and -o are inherently ambiguous.
+Use 'test EXPR1 && test EXPR2' or 'test EXPR1 || test EXPR2' instead.
```
Comment (sylvestre) [inline src/uu/test/test.md:70]:
```suggestion
NOTE: Binary `-a` and `-o` are inherently ambiguous.
Use `test EXPR1 && test EXPR2` or `test EXPR1 || test EXPR2` instead.
```

File: src/uu/test/test.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+NOTE: [ honors the --help and --version options, but test does not.
```
Comment (sylvestre) [inline src/uu/test/test.md:72]:
```suggestion
NOTE: `[` honors the `--help` and `--version` options, but test does not.
```

File: src/uu/test/test.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+test treats each of those as it treats any other nonempty `STRING`.
+
+NOTE: your shell may have its own version of test and/or [, which usually supersedes the version described here.
```
Comment (sylvestre) [inline src/uu/test/test.md:75]:
```suggestion
NOTE: your shell may have its own version of `test` and/or `[`, which usually supersedes the version described here.
```

File: src/uu/test/test.md (hunk 1 above)
Comment (m11o) [reply src/uu/test/test.md:25]: Thanks

File: src/uu/test/test.md (hunk 3 above)
Comment (sylvestre) [reply src/uu/test/test.md:40]: Please add * in the other places

File: src/uu/test/test.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+* EXPRESSION1 -a EXPRESSION2   both `EXPRESSION1` and `EXPRESSION2` are true
+* EXPRESSION1 -o EXPRESSION2   either `EXPRESSION1` or `EXPRESSION2` is true
+
+* -n STRING            the length of `STRING` is nonzero
+* STRING               equivalent to -n `STRING`
+* -z STRING            the length of `STRING` is zero
+* STRING1 = STRING2    the strings are equal
+* STRING1 != STRING2   the strings are not equal
+
+* INTEGER1 -eq INTEGER2   `INTEGER1` is equal to `INTEGER2`
```
Comment (sylvestre) [inline src/uu/test/test.md:33]:
```suggestion
Integer comparisons:
* INTEGER1 -eq INTEGER2   `INTEGER1` is equal to `INTEGER2`
```

File: src/uu/test/test.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+* INTEGER1 -ge INTEGER2   `INTEGER1` is greater than or equal to `INTEGER2`
+* INTEGER1 -gt INTEGER2   `INTEGER1` is greater than `INTEGER2`
+* INTEGER1 -le INTEGER2   `INTEGER1` is less than or equal to `INTEGER2`
+* INTEGER1 -lt INTEGER2   `INTEGER1` is less than `INTEGER2`
+* INTEGER1 -ne INTEGER2   `INTEGER1` is not equal to `INTEGER2`
+
+* FILE1 -ef FILE2   `FILE1` and `FILE2` have the same device and inode numbers
```
Comment (sylvestre) [inline src/uu/test/test.md:40]:
```suggestion
File operations:
* FILE1 -ef FILE2   `FILE1` and `FILE2` have the same device and inode numbers
```

**Pull request merged**